src/cart/src/cart/crt_rpc.c

```diff
@@ -178,13 +182,25 @@ crt_internal_rpc_register(void)
 	struct crt_proto_format	cpf;
 	int			rc;
 
-	cpf.cpf_name  = "internal-proto";
-	cpf.cpf_ver   = 0;
+	cpf.cpf_name  = "internal";
+	cpf.cpf_ver   = CRT_PROTO_INTERNAL_VERSION;
 	cpf.cpf_count = ARRAY_SIZE(crt_internal_rpcs);
 	cpf.cpf_prf   = crt_internal_rpcs;
 	cpf.cpf_base  = CRT_OPC_INTERNAL_BASE;
 
 	rc = crt_proto_register_internal(&cpf);
+	if (rc != 0) {
+		D_ERROR("crt_proto_register_internal() failed. rc %d\n", rc);
+		return rc;
+	}
+
+	cpf.cpf_name  = "fault-injection";
+	cpf.cpf_ver   = CRT_PROTO_FI_VERSION;
+	cpf.cpf_count = ARRAY_SIZE(crt_fi_rpcs);
+	cpf.cpf_prf   = crt_fi_rpcs;
+	cpf.cpf_base  = CRT_OPC_FI_BASE;
+
+	rc = crt_proto_register(&cpf);
```

Comment:
the error message below should be updated since call is now crt_proto_register().

Also if this fails i wonder if we should unroll/unregister the registration done on line 191 in this function

Reply:
Ok, will update.

It looks like if this function fails then crt_opc_map_destroy() gets called which would tear everything down, I'll check.